src/bgg/encoding.rs

```diff
@@ -164,33 +175,175 @@ impl<M: PolyMatrix> Evaluable for BggEncoding<M> {
         let plaintext = one.plaintext.clone().map(|plaintext| plaintext * const_poly);
         Self { vector, pubkey, plaintext }
     }
+
+    fn public_lookup(
+        self,
+        params: &Self::Params,
+        plt: &mut PublicLut<M>,
```

Comment:
Why does `plt` need to be mut?

Reply:
need to update the `A_z` that only can be called during evaluation of public key